usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateShapeIdReferenceInStopTimeTest.java

```diff
@@ -0,0 +1,293 @@
+
+        final ValidateShapeIdReferenceInStopTime underTest =
+                new ValidateShapeIdReferenceInStopTime(mockDataRepo, mockResultRepo, mockLogger);
+
+        underTest.execute();
+
+        verify(mockLogger, times(1)).info(ArgumentMatchers.eq(
+                "Validating rule 'E034 - `shape_id` must be provided and valid when" +
+                        " stop_times.shape_dist_travelled is provided'" + System.lineSeparator()));
+
+        verify(mockDataRepo, times(1)).getStopTimeAll();
+        verify(mockDataRepo, times(1))
+                .getStopTimeByTripId(ArgumentMatchers.eq("trip id"));
+        verify(mockDataRepo, times(1)).getTripById("trip id");
+        verify(mockDataRepo, times(1)).getShapeById("non existing shape id");
+
+        verify(mockStopTime, times(1)).getShapeDistTraveled();
+        verify(mockTrip, times(1)).getShapeId();
+
+        final ArgumentCaptor<NonExistingShapeNotice> captor =
+                ArgumentCaptor.forClass(NonExistingShapeNotice.class);
+
+        verify(mockResultRepo, times(1)).addNotice(captor.capture());
+
+        final List<NonExistingShapeNotice> noticeList = captor.getAllValues();
+
+        assertEquals("stop_times.txt", noticeList.get(0).getFilename());
+        assertEquals("shape_id", noticeList.get(0).getNoticeSpecific(KEY_FIELD_NAME));
+        assertEquals("trip_id", noticeList.get(0).getNoticeSpecific(KEY_COMPOSITE_KEY_FIRST_PART));
+        assertEquals("stop_sequence", noticeList.get(0).getNoticeSpecific(KEY_COMPOSITE_KEY_SECOND_PART));
+        assertEquals("trip id", noticeList.get(0).getNoticeSpecific(KEY_COMPOSITE_KEY_FIRST_VALUE));
+        assertEquals(2, noticeList.get(0).getNoticeSpecific(KEY_COMPOSITE_KEY_SECOND_VALUE));
+        assertEquals("non existing shape id", noticeList.get(0).getNoticeSpecific(KEY_UNKNOWN_SHAPE_ID));
+
+        verifyNoMoreInteractions(mockDataRepo, mockLogger, mockResultRepo, mockStopTime, mockTrip);
+    }
+
+    // suppress warning regarding ignored result of method since it is not necessary here.
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @Test
+    void stopTimeWithNonNullShapeDistTravelledPlusValidTripIdAndNullAndValidShapeIdShouldNotGenerateNotice() {
+        final StopTime mockStopTime = mock(StopTime.class);
+        when(mockStopTime.getTripId()).thenReturn("trip id");
+        when(mockStopTime.getStopSequence()).thenReturn(2);
+        when(mockStopTime.getShapeDistTraveled()).thenReturn(5f);
+        final Map<String, TreeMap<Integer, StopTime>> mockStopTimeCollection = new HashMap<>();
+        final TreeMap<Integer, StopTime> innerMap = new TreeMap<>();
+        innerMap.put(2, mockStopTime);
+        mockStopTimeCollection.put("trip id", innerMap);
+
+        final Trip mockTrip = mock(Trip.class);
+        when(mockTrip.getShapeId()).thenReturn(null);
+        when(mockTrip.getTripId()).thenReturn("trip id");
+        when(mockTrip.getShapeId()).thenReturn("non existing shape id");
```

Comment:
Same as before, should `when(mockTrip.getShapeId()).thenReturn();` implemented twice?

Reply:
Wrong `cmd+c - cmd+v` operation 😅